src/policy/immix/immixspace.rs

```diff
@@ -646,14 +721,25 @@ pub struct PrepareBlockState<VM: VMBinding> {
 
 impl<VM: VMBinding> PrepareBlockState<VM> {
     /// Clear object mark table
-    fn reset_object_mark(chunk: Chunk) {
+    fn reset_object_mark(&self) {
         // NOTE: We reset the mark bits because cyclic mark bit is currently not supported, yet.
         // See `ImmixSpace::prepare`.
         if let MetadataSpec::OnSide(side) = *VM::VMObjectModel::LOCAL_MARK_BIT_SPEC {
-            side.bzero_metadata(chunk.start(), Chunk::BYTES);
+            side.bzero_metadata(self.chunk.start(), Chunk::BYTES);
+        }
+        if self.space.space_args.reset_log_bit_in_major_gc {
+            if let MetadataSpec::OnSide(side) = *VM::VMObjectModel::GLOBAL_LOG_BIT_SPEC {
+                // We zero all the log bits in major GC, and for every object we trace, we will mark the log bit again.
+                side.bzero_metadata(self.chunk.start(), Chunk::BYTES);
```

Comment:
When do you need to bulk zero all the unlog bits in the heap? This sounds expensive.

Reply:
`PrepareBlockState` is only scheduled for full heap GCs. And this bulk zeroing is only for sticky immix, it does not affect other immix variants.

Reply:
Sorry I mean for sticky immix, why all the unlog table need to be reset before a full-heap GC?

Reply:
We need a way to set the log bit back to `0` for dead objects after a full heap GC. We can either clear the log bit when we allocate new objects in mutator phase, or just bulk zero the log bit and set the bit for the live objects. As I already need to set the bit for live objects (see our discussion in the other comment), I think it is natural to just bulk zero the log bit.